darshan-util/darshan-logutils.c

```diff
@@ -2143,7 +2143,10 @@ void darshan_log_get_name_records(darshan_fd fd,
     HASH_ITER(hlink, name_hash, curr, tmp)
     {
         (*name_records)[i].id = curr->name_record->id;
-        (*name_records)[i].name = curr->name_record->name;
```

Comment:
Good fix.  My only suggestion would be to put a comment in here explaining that the hash table was previously constructed on the fly by darshan_log_get_namehash(), so this function is responsible for releasing the hash table memory before it returns.  It doesn't look symmetric at first glance.

Reply:
Yeah, that's a good point. These functions might seem strange to someone at first glance, but really these are just helpers for PyDarshan that translate the name record hash Darshan would normally provide into something that can be managed by CFFI (an array of record ID, name tuples).

Reply:
This should be done in [8f7d8a3](https://github.com/darshan-hpc/darshan/pull/825/commits/8f7d8a318d7041469177fa2ef5b504c1b4e5da8b).

Reply:
perfect, thanks